miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppDisplay.kt

```diff
@@ -1,22 +1,35 @@
-package com.rakuten.tech.mobile.miniapp
+package com.rakuten.tech.mobile.miniapp.display
 
+import android.content.Context
 import android.view.View
+import androidx.lifecycle.Lifecycle
 import androidx.lifecycle.LifecycleObserver
+import com.rakuten.tech.mobile.miniapp.js.MiniAppMessageBridge
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
 
 /**
  * This represents the contract by which the host app can interact with the
  * display unit of the mini app.
- * This contract complies to Android's [LifecycleObserver] contract, and when made to observe
- * the lifecycle, it automatically clears up the view state and any services registered with.
  */
-interface MiniAppDisplay : LifecycleObserver {
+interface MiniAppDisplay {
+    val basePath: String
+    val appId: String
+    val miniAppMessageBridge: MiniAppMessageBridge
 
     /**
      * Provides the view associated with the mini app to the caller for showing the mini app.
+     * [LifecycleObserver] as optional. When made to observe the lifecycle,
+     * it automatically clears up the view state and any services registered with.
      * @return [View] as mini app's view with [LayoutParams] set to match
      * the parent's dimensions.
      */
-    fun getMiniAppView(): View
+    suspend fun getMiniAppView(context: Context, hostLifecycle: Lifecycle? = null): View =
```

Comment:
The argument `context` and `hostLifecycle` are not related here. The old method does not work in this situation because `hostLifecycle` observes the `RealMiniAppDisplay` which is created in this function `getMiniAppView(Context, LicycleObserver`).